File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.Arrays;
class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        boolean[] isSkip = new boolean[26];
        for(int i=0; i<skip.length(); i++){
            isSkip[skip.charAt(i) - 'a'] = true;
        }
        for(int i=0; i<s.length(); i++){
            int alphabetIndex = s.charAt(i) - 'a';
            int move = 0;
            while(move < index){
                if(isSkip[alphabetIndex]){
                    alphabetIndex = (alphabetIndex + 1) % 26;
                } else {
                    alphabetIndex = (alphabetIndex + 1) % 26;
                    move++;
                }
            }
            while(isSkip[alphabetIndex]){
                alphabetIndex = (alphabetIndex + 1) % 26;
            }
            answer += (char)('a' + alphabetIndex);
        }
        return answer;
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ''
    isSkip = [False] * 26
    for i in range(len(skip)):
        isSkip[ord(skip[i]) - ord('a')] = True
    for i in range(len(s)):
        alphabetIndex = ord(s[i]) - ord('a')
        move = 0
        while move < index:
            if isSkip[alphabetIndex]:
                alphabetIndex = (alphabetIndex + 1) % 26
            else:
                alphabetIndex = (alphabetIndex + 1) % 26
                move += 1
        while isSkip[alphabetIndex]:
            alphabetIndex = (alphabetIndex + 1) % 26
        answer += chr(alphabetIndex + ord('a'))
    return answer